src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Utility/VSRestoreSettingsUtilities.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NuGet.Commands;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.ProjectModel;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    public class VSRestoreSettingsUtilities
+    {
+        private static string RestoreAdditionalProjectFallbackFolders = nameof(RestoreAdditionalProjectFallbackFolders);
+        private static string RestoreAdditionalProjectSources = nameof(RestoreAdditionalProjectSources);
+
+        public static string GetPackagesPath(ISettings settings, PackageSpec project)
+        {
+            // Set from Settings if not given. Clear is not an option here.
+            if (string.IsNullOrEmpty(project.RestoreMetadata.PackagesPath))
+            {
+                return SettingsUtility.GetGlobalPackagesFolder(settings);
+            }
+
+            // Resolve relative paths
+            return UriUtility.GetAbsolutePathFromFile(
+                sourceFile: project.RestoreMetadata.ProjectPath,
+                path: project.RestoreMetadata.PackagesPath);
+        }
+
+        public static IList<PackageSource> GetSources(ISettings settings, PackageSpec project)
+        {
+            var sources = new List<string>();
+            var additionalSources = new List<string>();
+
+            var readingAdditionalSources = false;
+            foreach (var source in project.RestoreMetadata.Sources)
+            {
+                if (RestoreAdditionalProjectSources.Equals(source.Source))
```

Comment:
Good point. It shouldn't be causing any problems now because this is internal, but they don't depend on the same constant so it'd be safer to add the ignore case.